File: frontend/src/types/cashflow.ts
export type CashflowItem={id:string;data:string;tipo:string;origem:string;origem_id:string;descricao:string;categoria:string;valor:number;saldo_projetado:number;status:string}
export type CashflowProjection={summary:{current_balance:number;forecast_balance:number;month_inflows:number;month_outflows:number;net_result:number;tax_reserve:number;available:number};forecasts:{days:number;balance:number}[];series:{date:string;inflow:number;outflow:number;balance:number}[];insights:string[];alerts:string[]}
export type CashflowPageData={items:CashflowItem[];total:number;page:number;page_size:number}
export type CalendarData=Record<string,{id:string;description:string;value:number;balance:number;status:string}[]>

File: backend/app/schemas/cashflow.py
from datetime import date,datetime
from decimal import Decimal
import uuid
from pydantic import BaseModel,Field
class CashflowResponse(BaseModel):
 id:uuid.UUID;data:date;tipo:str;origem:str;origem_id:uuid.UUID;descricao:str;categoria:str;valor:Decimal;saldo_projetado:Decimal;status:str;created_at:datetime;updated_at:datetime
 model_config={'from_attributes':True}
class CashflowPage(BaseModel):items:list[CashflowResponse];total:int;page:int;page_size:int
class CashflowSimulation(BaseModel):
 delayed_origin_id:uuid.UUID|None=None;delay_days:int=Field(0,ge=0,le=365);extra_shifts:int=Field(0,ge=0,le=100);shift_value:Decimal=Field(0,ge=0);extra_expenses:Decimal=Field(0,ge=0);horizon_days:int=Field(90,ge=7,le=365)
